cmd/query/app/apiv3/grpc_handler.go

```diff
@@ -32,20 +34,24 @@ func (h *Handler) GetTrace(request *api_v3.GetTraceRequest, stream api_v3.QueryS
 		return fmt.Errorf("malform trace ID: %w", err)
 	}
 
-	query := querysvc.GetTraceParameters{
-		GetTraceParameters: spanstore.GetTraceParameters{
-			TraceID:   traceID,
-			StartTime: request.GetStartTime(),
-			EndTime:   request.GetEndTime(),
+	query := querysvc.GetTraceParams{
+		TraceIDs: []tracestore.GetTraceParams{
+			{
+				TraceID: traceID.ToOTELTraceID(),
+				Start:   request.GetStartTime(),
+				End:     request.GetEndTime(),
+			},
 		},
 		RawTraces: request.GetRawTraces(),
 	}
-	trace, err := h.QueryService.GetTrace(stream.Context(), query)
+	getTracesIter := h.QueryService.GetTraces(stream.Context(), query)
+	traces, err := iter.FlattenWithErrors(getTracesIter)
```

Comment:
the API returns a stream, so we should loop through iterator and write out chunks instead of collecting.